ion-schema/src/system.rs

```diff
@@ -754,37 +753,78 @@ impl Resolver {
         let mut isl_inline_imports: Vec<IslImportType> = vec![];
         let mut open_content = vec![];
         let mut isl_user_reserved_fields = UserReservedFields::default();
+        let mut isl_version = IslVersion::V1_0;
+        let reserved_keyword_version_marker =
+            Regex::new(r"^(\$ion_schema(_.*)?|[a-z][a-z0-9]*(_[a-z0-9]+)*)$").unwrap();
```

Comment:
Does this need a `OnceLock`?

Reply:
I recommended it. `OnceLock` is the thread-safe version of `OnceCell` that was just release in Rust 1.70, and it's a good way to move the compilation of the regex out of the regular codepath and into a separate initialization code path.

Reply:
Yes, I meant to add it but had missed this place.